Adapters/PerunRpcAdapter.py

```diff
@@ -0,0 +1,189 @@
+import logging
+from typing import List
+from models import User, Group, Vo, Facility, HasIdAbstract, Resource
+from utils import AttributeUtils
+
+from Adapters import AdapterInterface
+from perun_openapi.api.attributes_manager_api import AttributesManagerApi
+from perun_openapi.api.facilities_manager_api import FacilitiesManagerApi
+from perun_openapi.api.groups_manager_api import GroupsManagerApi
+from perun_openapi.api.members_manager_api import MembersManagerApi
+from perun_openapi.api.resources_manager_api import ResourcesManagerApi
+from perun_openapi.api.users_manager_api import UsersManagerApi
+from perun_openapi.api.vos_manager_api import VosManagerApi
+
+
+class PerunRpcAdapter(AdapterInterface):
+    def get_perun_user(self, idp_id: str, uids: List[str]) -> User:
+        for uid in uids:
+            user = UsersManagerApi.get_user_by_ext_source_name_and_ext_login(
+                self, uid, idp_id
+            )
+
+            name = ""
+            if user["titleBefore"]:
+                name += user["titleBefore"] + " "
+            if user["firstName"]:
+                name += user["firstName"] + " "
+            if user["middleName"]:
+                name += user["middleName"] + " "
+            if user["lastName"]:
+                name += user["lastName"]
+            if user["titleAfter"]:
+                name += " " + user["titleAfter"]
+
+            return User(user["id"], name)
+
+    def get_member_groups(self, user: User, vo: Vo) -> List[Group]:
+        member = MembersManagerApi.get_member_by_user(self, vo, user)
+        member_groups = GroupsManagerApi.get_all_member_groups(
+            self, member["id"]
+        )
+
+        converted_groups = []
+        for group in member_groups:
+            attr = AttributesManagerApi.get_attribute(group["id"])
+            unique_name = attr["value"] + ":" + group["name"]
+            converted_groups.append(
+                Group(
+                    group["id"],
+                    group["vo_id"],
+                    group["uuid"],
+                    group["name"],
+                    unique_name,
+                    group["description"],
+                )
+            )
+
+        return converted_groups
+
+    def get_sp_groups(self, sp_entity_id: str) -> List[Group]:
+
+        facility = self.get_facility_by_rp_identifier(sp_entity_id)
+        if facility is None:
+            return []
+
+        perun_attrs = FacilitiesManagerApi.get_assigned_resources_for_facility(
+            self, facility.id
+        )
+
+        resources = []
+        for perun_attr in perun_attrs:
+            resources.append(
+                Resource(
+                    perun_attr["id"],
+                    perun_attr["vo_id"],
+                    perun_attr["facility_id"],
+                    perun_attr["name"],
+                )
+            )
```

Comment:
```suggestion
        resources = [Resource(perun_attr["id"], perun_attr["vo_id"], perun_attr["facility_id"], perun_attr["name"]) for perun_attr in perun_attrs]
```
use list comprehension where possible